blog/2024-04-19-what-is-the-10101-coordinator.md

```diff
@@ -0,0 +1,81 @@
+---
+slug: what is the 10101 coordinator
+title: What is the 10101 coordinator?
+authors: [richard]
+tags: [DLC-Channels]
+---
+
+![](../static//2024-04-19-what-is-the-10101-coordinator/coordinator.png)
+
+10101 aims to bring self-custodial p2p perpetual futures trading to users without sacrificing UX. Centralized exchanges are easy as they only have to keep a book updating all events on their own system. But 10101 is decentralized. Every user is running their own trading service through the 10101 app or dlc connect web app.
+
+In order to compete with the simplicity of central exchange 10101 had to address the following challenges which are easy to solve on central exchanges, but harder to solve in a decentralized p2p trading solution.
+
+1. **Discovery**: Matching traders on opposite sides of a trade.
+1. **Control**: A trader remains in control to decide when his position is closed.
+1. **Speed**: A trade needs to be executed almost instantly.
+
+## What is the 10101 coordinator?
+
+The 10101 coordinator is the solution component to address the before mentioned UX challenges when delivering a self-custodial p2p trading solution.
+
+![](../static/2024-04-19-what-is-the-10101-coordinator/10101-network.png)
+
+The 10101 coordinator acts as a bridge to connect all 10101 traders with each other, while at the same time providing an 24/7 online service to coordinate trades.
+
+### Discovery
+
+On a central exchange a user is mererly a set of records in a database. A central system is keeping record of who opened what position at what price and handles the execution centrally. Clients are merely interfaces to interact with that central system, but the central system is in full control of all components at all time.
+
+At 10101 trades are setup between equal parties. A trade can only be setup if both nodes agree to the contract (discreet log contract). The 10101 coordinator simplifies that discovery by being the only node the users have to connect to. Through the coordinator any 10101 trader can discover another 10101 trader.
+
+### Control
+
+When submitting an order the coordinator will look for one or several matching counterparties. Once found the coordinator initiates the dlc protocol with each party. That means your direct counterparty will always be the 10101 coordinator, but implicitely whoever the coordinator matched you up with.
+
+For executing order all parties had to be online to exchange the required messages to establish a dlc. But once this is setup any 10101 Trader can independtly control their position, without having to rely on their indirect counterparty to be online.
+
```

Comment:
```suggestion

### Free Option Problem

The free option problem allows a trader to delay or back out of a trade after both users have collateralized it. He would do so based on whether the exchange rate has moved for or against their favor since the time that the terms were set.

Because DLC protocols are synchronous, one trader must be the first to sign all corresponding transactions. Trader 2 should do the same afterwards. This stepwise process gives the second user the unique ability to delay the trade while monitoring the prevailing exchange rate. If the rate has moved in their favor since the time that the terms were set (prevailing rate is now worse than the agreed-upon rate), then we should expect them to complete the swap. Conversely, if it has moved against them, it is in their best interest to let the swap expire, take their refund, and begin the process again.

This is called the **Free Option Problem**.

Centralized exchanges do not face this problem because the platform ensures safe execution. 

Similar in 10101 - the coordinator connects two traders with each other holding back both signatures until al signatures are available. This way, we solved the free option problem. 

```